File: opensrp-chw-referral/src/main/java/org/smartregister/chw/referral/presenter/BaseReferralRegisterPresenter.java
package org.smartregister.chw.referral.presenter;

import android.support.annotation.VisibleForTesting;

import org.apache.commons.lang3.tuple.Triple;
import org.smartregister.chw.referral.contract.BaseReferralRegisterContract;

import java.lang.ref.WeakReference;
import java.util.List;

public class BaseReferralRegisterPresenter implements BaseReferralRegisterContract.Presenter, BaseReferralRegisterContract.InteractorCallBack {
    protected WeakReference<BaseReferralRegisterContract.View> viewReference;
    protected BaseReferralRegisterContract.Model model;

    public BaseReferralRegisterPresenter(BaseReferralRegisterContract.View view, BaseReferralRegisterContract.Model model) {
        viewReference = new WeakReference<>(view);
        this.model = model;
    }

    @Override
    public void saveLanguage(String language) {
//        implement
    }

    @Override
    public void closeFamilyRecord(String jsonString) {
//        implement
    }

    @Override
    public void onUniqueIdFetched(Triple<String, String, String> triple, String entityId) {
//        implement
    }

    @Override
    public void onNoUniqueId() {
//        implement
    }

    @Override
    public void onRegistrationSaved() {
        getView().hideProgressDialog();
    }

    @Override
    public void registerViewConfigurations(List<String> list) {
//        implement
    }

    @Override
    public void unregisterViewConfiguration(List<String> list) {
//        implement
    }

    @Override
    public void onDestroy(boolean b) {
//        implement
    }

    @Override
    public void updateInitials() {
//        implement
    }

    @VisibleForTesting
    BaseReferralRegisterContract.View getView() {
        if (viewReference != null)
            return viewReference.get();
        else
            return null;
    }
}

File: opensrp-chw-referral/src/main/java/org/smartregister/chw/referral/presenter/BaseReferralRegisterPresenter.kt
package org.smartregister.chw.referral.presenter

import org.smartregister.chw.referral.contract.BaseReferralRegisterContract
import java.lang.ref.WeakReference

open class BaseReferralRegisterPresenter(
    view: BaseReferralRegisterContract.View,
    protected var model: BaseReferralRegisterContract.Model
) : BaseReferralRegisterContract.Presenter {

    private var viewReference = WeakReference(view)

    override fun getView() = viewReference.get()

    override fun registerViewConfigurations(list: List<String>) = Unit

    override fun unregisterViewConfiguration(list: List<String>) = Unit

    override fun onDestroy(b: Boolean) = Unit

    override fun updateInitials() = Unit

}